contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3560,6 +3615,22 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 						else
 							orig_schema = "dbo";
 					}
+					else
+					{
+						owner_oid = rolspec ? get_rolespec_oid(rolspec, true) : InvalidOid;
+						if (OidIsValid(owner_oid) && OidIsValid(db_accessadmin) && !member_can_set_role(GetUserId(), owner_oid) &&
+							has_privs_of_role(GetUserId(), db_accessadmin) &&
+							(is_user(owner_oid, true) || is_role(owner_oid, true)))
```

Comment:
What happens if owner_oid is invalid? It looks like it will allow the creation with current user as owner. Please correct me if it's not the case.

Reply:
Do you mean the first check `OidIsValid(owner_oid)` is not enough ?

Reply:
Let's add a comment regarding what this condition is checking.

Reply:
Done